File: app/src/main/java/com/example/nathan/loadtracker/arrayadapters/JobSessionArrayAdapter.java
package com.example.nathan.loadtracker.arrayadapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;

import com.example.nathan.loadtracker.Item;

import java.util.List;

public class JobSessionArrayAdapter extends ArrayAdapter<Item> {

    private LayoutInflater mInflater;

    public enum RowType {
        LIST_ITEM, HEADER_ITEM
    }

    public JobSessionArrayAdapter(Context context, List<Item> items) {
        super(context, 0, items);
        this.mInflater = LayoutInflater.from(context);
    }

    public int getItemViewType(int position) {
        return getItem(position).getViewType();
    }

    public View getView(int position, View convertView, ViewGroup parent) {
        return getItem(position).getView(mInflater, convertView);
    }
}

File: app/src/main/java/com/example/nathan/loadtracker/arrayadapters/JobSessionArrayAdapter.kt
package com.example.nathan.loadtracker.arrayadapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter

import com.example.nathan.loadtracker.Item

class JobSessionArrayAdapter(context: Context, items: List<Item>) : ArrayAdapter<Item>(context, 0, items) {

    private val mInflater: LayoutInflater = LayoutInflater.from(context)

    enum class RowType {
        LIST_ITEM, HEADER_ITEM
    }

    override fun getItemViewType(position: Int): Int {
        return getItem(position)!!.viewType
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        return getItem(position)!!.getView(mInflater, convertView)
    }
}
